src/arena.c

```diff
@@ -3772,11 +3782,78 @@ bin_info_init(void)
 #undef SC
 }
 
+static void
+init_thp_initially_huge(void) {
+	int fd;
+	char buf[sizeof("[always] madvise never\n")];
+	ssize_t nread;
+	static const char *enabled_states[] = {
+		"[always] madvise never\n",
+		"always [madvise] never\n",
+		"always madvise [never]\n"
```

Comment:
In practice that's what the code currently does, because we start off with ```thp_initially_huge``` set to false, and therefore never call ```pages_nohuge()```.